mlrun/runtimes/funcref.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import mlrun
+
+from ..model import ModelObj
+from ..utils import generate_object_uri
+
+
+class FunctionRef(ModelObj):
+    """function reference/template"""
+
+    def __init__(
+        self,
+        url=None,
+        image=None,
+        requirements=None,
+        code=None,
+        spec=None,
+        kind=None,
+        name=None,
```

Comment:
Can you elaborate why would a function reference have anything else than... the reference which is the URL, what's the motivation ?

Reply:
this is more of a template, not a full func, and user want to be able to specify/load from a simple py file/notebook, not forced to create a function spec, we do the same in project (we can use this object under project as well), the spec part allows us to place full func, or part that we want to add (env/volumes/gpu). so we need a way to specify the image/kind and requirements.